honestcash-v2/src/app/modules/editor/converters/json-to-html.ts

```diff
@@ -156,6 +156,13 @@ export const convertBlockToHtml = (block: Block): string | null => {
       return convertListBlockToHtml(<ListElement>block);
     }
     case 'image': {
+      // hide images that don't have a url so that it does not appear as broken
```

Comment:
If the link is broken, optimally the block is shown to user and the user can delete it on its own. Otherwise the block stays in the JSON, right?

Reply:
the block is shown to the user in the editor however upon save if the user does save it with no uploaded image (if it failed or he did not upload at all) or that the image is broken it will still be saved but on stories and paywalls it will create problems:

on paywalls if there is a broken link and while you select the linebreak, as the image has nothing to show visually, the image does not even appear as broken image icon on the paywlal selection thus you can select your post wrong because in the end story will not show that image (block converter function will convert it, it will get inserted into dom as img but just blank space area in story so kind of weird)

so on the backend we should filter any images that have url: "" and remove them if the user did not (at publish stage because at save stage user can delete that block on another save before publish)

Reply:
Ok, sounds legit. Thank you for the explanation.